app/src/handlers/PlayerHandler.ts

```diff
@@ -0,0 +1,35 @@
+import {injectable} from 'inversify';
+import PlayerHandlerInterface from '../interfaces/PlayerHandlerInterface';
+import PlayerLoginEvent from './census/events/PlayerLoginEvent';
+import PlayerLogoutEvent from './census/events/PlayerLogoutEvent';
+
+declare type Player = {
+    worldId: number;
+    characterId: number;
+    outfitId: number;
+};
+
+@injectable()
+export default class PlayerHandler implements PlayerHandlerInterface {
+    private _players: Player[] = [];
+
+    private containsPlayer(worldId: number, characterId: number) {
+        return this._players.some((player) => {
+            return player.worldId === worldId && player.characterId === characterId;
+        });
+    }
+    handleLogin(loginEvent: PlayerLoginEvent): boolean {
+        if (!this.containsPlayer(loginEvent.worldId, loginEvent.characterId)) {
+            this._players.push({worldId:loginEvent.worldId, characterId:loginEvent.characterId, outfitId:-1});
+            return true;
+        }
+        return false;
+    }
+
+    handleLogout(logoutEvent: PlayerLogoutEvent): boolean {
+        this._players = this._players.filter((player) => {
+            return (player.worldId !== logoutEvent.worldId || player.characterId !== logoutEvent.characterId);
+        });
+        return true;
```

Comment:
You want to do this in the first implementation here?

Reply:
Sure, can we do loops within classes in this context? Guessing we have to "boot" the class? Could always do it as a service also.

Reply:
A background task maybe better?
May cost some performance to iterate over 3k players to often?

Reply:
I wouldn't imangine it would cost much to do a loop for 3k entries, 30k maybe, not not necessarily 3k. 

@microwavekonijn would this background task be it's own service architecturally?

Reply:
Moving this to seperate issue.

Need a background task to prune the online players for the PS2 worlds. We also need to update handlers to keep this information fresh as possible, using a "last seen" unix timestamp. The task will run through the online players and check their last online times, and if it's over a certain threshhold (currently thinking 10 mins) remove them from being online.

Reply:
I don't think it needs to be its own service. However I would put a list like that into it's own class something like DecayingSet and use that. With some clever use of a WeakSet and a rotating array of normal Sets you can create better performance without having to iterate over all elements.

In general use Set and Map over Array and Object in these kind of cases. They have better performance as I believe it implements a heap under the hood(though don't quote me on that). Also WeakSet and WeakMap are pretty cool as they do not block the garbage collector from freeing memory.